src/org/jetbrains/plugins/scala/lang/psi/types/ScTypePsiTypeBridge.scala

```diff
@@ -169,6 +171,11 @@ trait ScTypePsiTypeBridge {
       case ScCompoundType(Seq(typez, _*), _, _) => toPsi(typez, project, scope)
       case ScDesignatorType(c: ScTypeDefinition) if ScType.baseTypesQualMap.contains(c.qualifiedName) =>
         toPsi(ScType.baseTypesQualMap.get(c.qualifiedName).get, project, scope, noPrimitives, skolemToWildcard)
+      case ScDesignatorType(valType: ScClass) if isValueType(valType) && !noPrimitives =>
```

Comment:
So the reason I did it was for the following case:

```
class DoubleWrapper(val d: Double)
```

Then Foo[DoubleWrapper] => Foo[DoubleWrapper] (NOT Foo[Double], which is wrong because it would be converted to Java as Foo<double>)

But you are right about Wrappers on non-primitive types.